packages/cdktf-cli/lib/project-execution.ts

```diff
@@ -0,0 +1,448 @@
+                synthesizedStacks: (_context, event) => event.data,
+              }),
+            },
+          ],
+        },
+      },
+      diff: {
+        invoke: {
+          id: "diff",
+          src: "diff",
+          onError: {
+            target: "error",
+            actions: assign({
+              message: (_context, event) => {
+                const errorMessage = event.data.stderr || event.data;
+                return `terraform plan errored with: \n${errorMessage}`;
+              },
+            }),
+          },
+          onDone: [
+            {
+              target: "done",
+              actions: assign({
+                targetStackPlan: (_context, event) => event.data,
+              }),
+              cond: {
+                type: "onTargetAction",
+                name: "actionIsDiff",
+                value: "diff",
+              },
+            },
+
+            {
+              actions: assign({
+                targetStackPlan: (_context, event) => event.data,
+              }),
+              target: "gatherOutput",
+              cond: "planNeedsNoApply",
+            },
+            {
+              actions: assign({
+                targetStackPlan: (_context, event) => event.data,
+              }),
+              target: "approved",
+              cond: "autoApprove",
+            },
+            {
+              actions: assign({
+                targetStackPlan: (_context, event) => event.data,
+              }),
+              target: "waitingForApproval",
+            },
+          ],
+        },
+      },
+      waitingForApproval: {
+        on: {
+          APPROVAL_ABORTED: {
+            target: "done",
+          },
+          APPROVAL_GIVEN: {
+            target: "approved",
+          },
+        },
+      },
+      approved: {
+        always: [
+          {
+            target: "deploy",
+            cond: {
+              type: "onTargetAction",
+              name: "actionIsDiff",
+              value: "deploy",
+            },
+          },
+          {
+            target: "destroy",
+            cond: {
+              type: "onTargetAction",
+              name: "actionIsDiff",
+              value: "destroy",
+            },
+          },
+        ],
+      },
+      deploy: {
+        invoke: {
+          id: "deploy",
+          src: "deploy",
+          onError: {
+            target: "error",
+            actions: assign({
+              message: (_context, event) => event.data,
+            }),
+          },
+          onDone: {
+            target: "gatherOutput",
+          },
+        },
+      },
+      destroy: {
+        invoke: {
+          id: "destroy",
+          src: "destroy",
+          onError: {
+            target: "error",
+            actions: assign({
+              message: (_context, event) => event.data,
+            }),
+          },
+          onDone: {
+            target: "gatherOutput",
+          },
+        },
+      },
+      gatherOutput: {
```

Comment:
I'd like plural more as that differentiates it from "output" as in cli / shell output and brings "Terraform outputs" closer 🙂
```suggestion
      gatherOutputs: {
```